src/Socket.c

```diff
@@ -521,14 +521,14 @@ SOCKET Socket_getReadySocket(int more_work, int timeout, mutex_type mutex, int*
 			else
 				mod_s.saved.fds_write = malloc(mod_s.nfds * sizeof(struct pollfd));
 		}
-		memcpy(mod_s.saved.fds_read, mod_s.fds_read, mod_s.nfds * sizeof(struct pollfd));
-		memcpy(mod_s.saved.fds_write, mod_s.fds_write, mod_s.nfds * sizeof(struct pollfd));
-
-		if (mod_s.saved.nfds == 0)
+		
+		if ( mod_s.fds_read == NULL || mod_s.fds_write == NULL)
```

Comment:
We definitely need to make sure the memory allocations worked, but then we'll need to return an error rather than just "no work to do". Probably PAHO_MEMORY_ERROR.

Reply:
sock = PAHO_MEMORY_ERROR;